app/models/price_policy.rb

```diff
@@ -32,6 +32,17 @@ def self.current
     current_for_date(Time.zone.now)
   end
 
+  def self.current_and_newest
+    # TODO: Fix bug that causes this to be neccessary (in truncate_existing_policies)
+    # This method returns the newest price policy for when price policies accidentally overlap.
+    current.newest
+  end
+
+  def self.newest
+    ids = group(:price_group_id).maximum(:id).collect(&:last)
```

Comment:
I don't have a strong preference, but I had to try out `group(:price_group_id).maximum(:id)` to see that it returns a hash of `{ price_group_id => max_id }`. We could call `values` on it. Might read a little clearer than the `collect` and make it more obvious what the earlier methods return.